### PR TITLE
docs: make atomic-automatic-facet properties internal

### PR DESCRIPTION
**Pull request by alexprudhomme**
https://coveord.atlassian.net/browse/DOC-13363

The documentation team received a request to document better the description of the properties of the atomic-automatic-facet. However, atomic-automatic-facet are **never** supposed to be used by a user. The atomic-automatic-facet-generator generates the facets atomic-automatic-facet. The only reason why the component is not internal as a whole is because they can modify the css parts.

This PR also better shows that 5 is the default value for desiredCount.

### CODE REVIEW

File: packages/atomic/src/components/search/facets/atomic-automatic-facet/atomic-automatic-facet.tsx [hunk 1]
```diff
@@ -17,12 +17,11 @@ import {Bindings} from '../../atomic-search-interface/atomic-search-interface';
...
  * An automatic facet is a special type of facet generated by the automatic facets feature.
+ * Unlike traditional facets that need to be explicitly defined and requested in the query,
+ * automatic facets are dynamically generated by the index in response to the query.
  *
- * Unlike traditional facets that need to be explicitly defined and requested in the query, automatic facets are dynamically generated by the index
- * in response to the query.
- *
- * It's important to note that this component is not typically used on its own.
- * It is internally used by the `atomic-automatic-facet-generator` component to automatically render updated facets.
+ * **Note:** This component should never be used on it's own. It's used internally by the `atomic-automatic-facet-generator`
```
Comment (fbeaudoincoveo) [inline packages/atomic/src/components/search/facets/atomic-automatic-facet/atomic-automatic-facet.tsx:23]:
```suggestion
 * **Note:** This component should never be used on its own. It iss used internally by the `atomic-automatic-facet-generator`
```

File: packages/atomic/src/components/search/facets/atomic-automatic-facet/atomic-automatic-facet.tsx [hunk 2]
```diff
@@ -17,12 +17,11 @@ import {Bindings} from '../../atomic-search-interface/atomic-search-interface';
...
+ * component to automatically render updated facets. However, you can use the shadow parts to style the automatically generated facets.
  *
  * To read more about the automatic facet generator feature, see: https://docs.coveo.com/en/n9sd0159/
```
Comment (fbeaudoincoveo) [inline packages/atomic/src/components/search/facets/atomic-automatic-facet/atomic-automatic-facet.tsx:26]:
```suggestion
 * To learn more about the automatic facet generator feature, see: https://docs.coveo.com/en/n9sd0159/
```

File: packages/atomic/src/components/search/facets/atomic-automatic-facet/atomic-automatic-facet.tsx [hunk 3]
```diff
@@ -17,12 +17,11 @@ import {Bindings} from '../../atomic-search-interface/atomic-search-interface';
 
 /**
  * An automatic facet is a special type of facet generated by the automatic facets feature.
+ * Unlike traditional facets that need to be explicitly defined and requested in the query,
```
Comment (fbeaudoincoveo) [inline packages/atomic/src/components/search/facets/atomic-automatic-facet/atomic-automatic-facet.tsx:20]:
```suggestion
 * Unlike regular facets that need to be explicitly defined and requested in the query,
```

File: packages/atomic/src/components.d.ts [hunk 4]
```diff
@@ -67,11 +67,11 @@ export namespace Components {
...
-     * Unlike traditional facets that need to be explicitly defined and requested in the query, automatic facets are dynamically generated by the index
-     * in response to the query.
-     * It's important to note that this component is not typically used on its own.
-     * It is internally used by the `atomic-automatic-facet-generator` component to automatically render updated facets.
-     * To read more about the automatic facet generator feature, see: https://docs.coveo.com/en/n9sd0159/
+     * Unlike regular facets that need to be explicitly defined and requested in the query,
+     * automatic facets are dynamically generated by the index in response to the query.
+     * **Note:** This component should never be used on its own. It is used internally by the `atomic-automatic-facet-generator`
+     * component to automatically render updated facets. However, you can use the shadow parts to style the automatically generated facets.
+     * To learn more about the automatic facet generator feature, see: https://docs.coveo.com/en/n9sd0159/
```
Comment (srosenberg-coveo) [inline packages/atomic/src/components.d.ts:74]:
```suggestion
     * To learn more about the automatic facet generator feature, see: [About the Facet Generator](https://docs.coveo.com/en/n9sd0159/).
```

File: packages/atomic/src/components.d.ts [hunk 5]
```diff
@@ -2558,11 +2558,11 @@ declare global {
...
```
Comment (srosenberg-coveo) [inline packages/atomic/src/components.d.ts:2565]:
```suggestion
     * To learn more about the automatic facet generator feature, see: [About the Facet Generator](https://docs.coveo.com/en/n9sd0159/).
```

File: packages/atomic/src/components.d.ts [hunk 6]
```diff
@@ -3956,11 +3956,11 @@ declare namespace LocalJSX {
...
```
Comment (srosenberg-coveo) [inline packages/atomic/src/components.d.ts:3963]:
```suggestion
     * To learn more about the automatic facet generator feature, see: [About the Facet Generator](https://docs.coveo.com/en/n9sd0159/).
```

File: packages/atomic/src/components.d.ts [hunk 7]
```diff
@@ -6428,11 +6428,11 @@ declare module "@stencil/core" {
...
-             * Unlike traditional facets that need to be explicitly defined and requested in the query, automatic facets are dynamically generated by the index
-             * in response to the query.
-             * It's important to note that this component is not typically used on its own.
-             * It is internally used by the `atomic-automatic-facet-generator` component to automatically render updated facets.
-             * To read more about the automatic facet generator feature, see: https://docs.coveo.com/en/n9sd0159/
+             * Unlike regular facets that need to be explicitly defined and requested in the query,
+             * automatic facets are dynamically generated by the index in response to the query.
+             * **Note:** This component should never be used on its own. It is used internally by the `atomic-automatic-facet-generator`
+             * component to automatically render updated facets. However, you can use the shadow parts to style the automatically generated facets.
+             * To learn more about the automatic facet generator feature, see: https://docs.coveo.com/en/n9sd0159/
```
Comment (srosenberg-coveo) [inline packages/atomic/src/components.d.ts:6435]:
```suggestion
             * To learn more about the automatic facet generator feature, see: [About the Facet Generator](https://docs.coveo.com/en/n9sd0159/).
```

File: packages/atomic/src/components/search/facets/atomic-automatic-facet/atomic-automatic-facet.tsx [hunk 8]
```diff
@@ -17,14 +17,13 @@ import {Bindings} from '../../atomic-search-interface/atomic-search-interface';
...
- * Unlike traditional facets that need to be explicitly defined and requested in the query, automatic facets are dynamically generated by the index
- * in response to the query.
+ * **Note:** This component should never be used on its own. It is used internally by the `atomic-automatic-facet-generator`
+ * component to automatically render updated facets. However, you can use the shadow parts to style the automatically generated facets.
  *
- * It's important to note that this component is not typically used on its own.
- * It is internally used by the `atomic-automatic-facet-generator` component to automatically render updated facets.
- *
- * To read more about the automatic facet generator feature, see: https://docs.coveo.com/en/n9sd0159/
+ * To learn more about the automatic facet generator feature, see: https://docs.coveo.com/en/n9sd0159/
```
Comment (srosenberg-coveo) [inline packages/atomic/src/components/search/facets/atomic-automatic-facet/atomic-automatic-facet.tsx:26]:
```suggestion
 * To learn more about the automatic facet generator feature, see: [About the Facet Generator](https://docs.coveo.com/en/n9sd0159/).
```

**Review by srosenberg-coveo — APPROVED** [review]
One minor suggestion to make the link markdown. Otherwise, looks great!

File: packages/atomic/src/components/search/facets/atomic-automatic-facet-generator/atomic-automatic-facet-generator.tsx [hunk 9]
```diff
@@ -18,15 +18,13 @@ import {Bindings} from '../../atomic-search-interface/atomic-search-interface';
...
+ * and requested in the query, automatic facets are dynamically generated by the index
  * in response to the search query.
  *
- * It dynamically creates multiple `atomic-automatic-facet`
- * components based on the desiredCount prop.
+ * **Note:** This component renders multiple `atomic-automatic-facet` based on the desiredCount property.
  *
- * To read more about the automatic facet generator feature, see: https://docs.coveo.com/en/n9sd0159/
+ * To learn more about the automatic facet generator feature, see: [About the Facet Generator](https://docs.coveo.com/en/n9sd0159/).
  */
```
Comment (alexprudhomme) [inline packages/atomic/src/components/search/facets/atomic-automatic-facet-generator/atomic-automatic-facet-generator.tsx:28]: @srosenberg-coveo hey sam, I changed also the description in the generator description. Do you have any comments? I am not sure about the 25th line.

**Pull request merged**